hw2/04-api.js

```diff
@@ -1,7 +1,31 @@
 /** Exercise 04 - API **/
+//using v2 API which is already sorted
+const url = 'https://restcountries.com/v2/all';
+//for formatting population
+const nf = new Intl.NumberFormat('en-US');
 
-const url = 'https://restcountries.com/v3.1/all';
+let app = document.querySelector('#results');
+const getData = async function (url) {
+    try {
+        let response = await fetch(url);
+        let data = await response.json();
 
-// Add your code here
+        data.forEach((item) => {
+            addCountryAndPopulationToDOM(item);
+        });
+    } catch (error) {
+        console.error(error);
+        let element = document.createElement('div');
+        element.textContent = 'An error occured. Please try again';
+        app.append(element);
+    }
+};
+const addCountryAndPopulationToDOM = (item) => {
+
+    let li = document.createElement('li');
+    const population = isNaN(item.population) ? "Invalid population" : nf.format(item.population);
```

Comment:
I like how you used isNaN to validate the population just in case the API sends a wrong format.